src/components/FormProduct/index.jsx

```diff
@@ -14,88 +14,139 @@ export const FormProduct = () => {
     numberOfPurchases: 0,
     estimatedDaysNextPurchase: null,
   };
-
   const [product, setProduct] = useState(initialStateProduct);
-
-  const [error, setError] = useState('');
-  const handleInputProduct = (e) => {
-    setProduct({
-      ...product,
-      [e.target.name]: e.target.value,
-    });
-    setError(
-      e.target.value ? (
-        ''
-      ) : (
-        <span aria-live="polite">All fields are required</span>
-      ),
-    );
+  const [error, setError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState(false);
+  const [successSave, setSuccessMsg] = useState(false);
+  const handleNameInput = (e) => {
+    const clone = { ...product };
+    clone.item = e.target.value;
+    setProduct(clone);
+    setError(false);
+    setErrorMsg('');
+    setSuccessMsg(false);
   };
+  const handleRadioNextPurchase = (e) => {
+    const clone = { ...product };
+    clone.nextPurchase = e.target.value;
+    setProduct(clone);
+    setError(false);
+    setErrorMsg('');
+    setSuccessMsg(false);
+  };
+
   const handleSubmitProduct = (e) => {
     e.preventDefault();
     const { item, nextPurchase } = product;
 
-    if (isProductDuplicated(products, item))
-      return setError('The item is already on the list');
-
-    const editedProduct = {
-      ...product,
-      nextPurchase: Number(nextPurchase),
-      creationDate: new Date(),
-    };
-    addProduct(editedProduct);
-    setProduct({ ...initialStateProduct });
+    if (item.length > 0) {
+      if (nextPurchase > 0) {
+        if (isProductDuplicated(products, item)) {
+          setError(true);
+          setErrorMsg('The item is already on the list');
+        } else {
+          const editedProduct = {
+            ...product,
+            nextPurchase: Number(nextPurchase),
+            creationDate: new Date(),
+          };
+          addProduct(editedProduct);
+          setProduct({ ...initialStateProduct });
+          setError(false);
+          setErrorMsg('');
+          setSuccessMsg(true);
+        }
+      } else {
```

Comment:
I find it odd that you are using a single message. 
If the app is missing both the name and next purchase date, you want both messages to show at the same time. 
You don't want the user to click 'Add Item' for every single field to notice that it is required. 
No input -> submit -> Error: You are missing the name
User adds the name -> submit -> Error: You are missing next purchase date
User adds the purchase date -> submit -> success

Ideally:
No input -> Submit -> Error: You are missing name (next to name), You are missing next purchase date 
User adds both, the name and next purchase date -> Success!

Reply:
Thanks Indira! I appreciate your comment and observation. Yes, it's a better solution. Fixed.

![image](https://user-images.githubusercontent.com/8689897/109531069-d0621b80-7a8d-11eb-9444-5615d3c24835.png)
![image](https://user-images.githubusercontent.com/8689897/109531158-ef60ad80-7a8d-11eb-8697-f028068c9dad.png)
![image](https://user-images.githubusercontent.com/8689897/109531205-fdaec980-7a8d-11eb-8d3a-79ac9908ac4d.png)
![image](https://user-images.githubusercontent.com/8689897/109531294-17501100-7a8e-11eb-8fe3-f5cb7ba11171.png)